File: LC/Python/Easy/1317.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        left, right = 0, 0
        rounds = 0

        while n > 0:
            temp = n%10
            n = n//10

            if temp <= 1 and n > 0:
                temp += 10
                n -= 1

            remain = temp%2
            left += (temp//2)*10**rounds
            right += (temp//2+remain)*10**rounds
            rounds += 1
        
        return [left, right]

File: LC/C++/Easy/1317.cpp
class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        int left = 0, right = 0;
        int place = 1;
        
        while (n > 0) {
            int temp = n % 10;
            n /= 10;

            if (temp <= 1 && n > 0) {
                temp += 10;
                n--;
            }

            int remain = temp % 2;
            left += (temp / 2) * place;
            right += (temp / 2 + remain) * place;

            place *= 10;
        }

        return {left, right};
    }
};